File: frontend/src/services/jobListing.ts
import api from './api'

export interface JobListing {
  id: number
  job_title: string
  company_name: string
  salary_range?: string
  salary_avg?: number
  location?: string
  openings?: number
  requirements?: string
  search_keyword?: string
  data_year?: number
  created_at: string
}

export interface JobListingListResponse {
  items: JobListing[]
  total: number
  skip: number
  limit: number
  page: number
  pages: number
}

export interface JobListingQuery {
  skip?: number
  limit?: number
  job_title?: string
  company_name?: string
  location?: string
  search_keyword?: string
  data_year?: number
}

export interface SalaryStatistics {
  years: number[]
  categories: string[]
  data: Record<number, Record<string, number>>
}

export interface LocationStatistics {
  years: number[]
  provinces: string[]
  data: Record<number, Record<string, number>>
}

export interface WordCloudItem {
  name: string
  value: number
}

export interface WordCloudStatistics {
  years: number[]
  wordcloud: WordCloudItem[]
  year_statistics: Record<number, Record<string, number>>
  total_words: number
  unique_words: number
}

export const jobListingService = {
  // 获取招聘信息列表
  async getJobListings(params: JobListingQuery = {}): Promise<JobListingListResponse> {
    return api.get('/job-listings/', { params })
  },

  // 获取单个招聘信息
  async getJobListing(id: number): Promise<JobListing> {
    return api.get(`/job-listings/${id}`)
  },

  // 获取薪资统计数据
  async getSalaryStatistics(startYear: number = 2022, endYear: number = 2025): Promise<SalaryStatistics> {
    return api.get('/job-listings/salary-statistics', {
      params: { start_year: startYear, end_year: endYear }
    })
  },

  // 获取地理分布统计数据
  async getLocationStatistics(startYear: number = 2022, endYear: number = 2025): Promise<LocationStatistics> {
    return api.get('/job-listings/location-statistics', {
      params: { start_year: startYear, end_year: endYear }
    })
  },

  // 获取词云统计数据
  async getWordCloudStatistics(
    startYear: number = 2022, 
    endYear: number = 2025,
    topN: number = 100,
    minLength: number = 2
  ): Promise<WordCloudStatistics> {
    return api.get('/job-listings/wordcloud-statistics', {
      params: { 
        start_year: startYear, 
        end_year: endYear,
        top_n: topN,
        min_length: minLength
      }
    })
  },
}

File: app/schemas/job_listing.py
"""
招聘信息数据模式
"""
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime


class JobListingBase(BaseModel):
    """招聘信息基础模式"""
    job_title: str = Field(..., max_length=250, description="职位名称")
    company_name: str = Field(..., max_length=250, description="公司名称")
    salary_range: Optional[str] = Field(None, max_length=100, description="薪资范围")
    salary_avg: Optional[int] = Field(None, description="平均月薪（元）")
    location: Optional[str] = Field(None, max_length=100, description="工作地点")
    openings: Optional[int] = Field(None, ge=0, description="招聘人数")
    requirements: Optional[str] = Field(None, description="职位要求")
    search_keyword: Optional[str] = Field(None, max_length=100, description="搜索关键词")
    data_year: Optional[int] = Field(None, description="数据年份")


class JobListingCreate(JobListingBase):
    """创建招聘信息模式"""
    pass


class JobListingUpdate(BaseModel):
    """更新招聘信息模式"""
    job_title: Optional[str] = Field(None, max_length=250, description="职位名称")
    company_name: Optional[str] = Field(None, max_length=250, description="公司名称")
    salary_range: Optional[str] = Field(None, max_length=100, description="薪资范围")
    salary_avg: Optional[int] = Field(None, description="平均月薪（元）")
    location: Optional[str] = Field(None, max_length=100, description="工作地点")
    openings: Optional[int] = Field(None, ge=0, description="招聘人数")
    requirements: Optional[str] = Field(None, description="职位要求")
    search_keyword: Optional[str] = Field(None, max_length=100, description="搜索关键词")
    data_year: Optional[int] = Field(None, description="数据年份")


class JobListing(JobListingBase):
    """招聘信息响应模式"""
    id: int
    created_at: datetime
    
    class Config:
        from_attributes = True


class JobListingListResponse(BaseModel):
    """招聘信息列表响应模式（包含分页信息）"""
    items: List[JobListing] = Field(..., description="招聘信息列表")
    total: int = Field(..., description="总记录数")
    skip: int = Field(..., description="跳过记录数")
    limit: int = Field(..., description="每页记录数")
    page: int = Field(..., description="当前页码")
    pages: int = Field(..., description="总页数")

